src/main/webapp/app/tasks/local-task.service.ts

```diff
@@ -16,25 +16,43 @@ export class LocalTaskService implements TaskService {
 
   create(name: string): Observable<Task> {
     const tasks = this.readTasks();
-    const task = {id: uuid(), name};
+    // Added new key value pair
+    const task = {id: uuid(), name, isCompleted: false};
     tasks.push(task);
     this.writeTasks(tasks);
     return of(task);
   }
 
+  update(id: string, task: Task): Observable<Task> {
+    const tasks = this.readTasks();
+    const index = tasks.findIndex(t => t.id === id);
+
+    if (index !== -1) {
+      task.isCompleted = !task.isCompleted;
+      tasks[index].isCompleted = !tasks[index].isCompleted;
+      this.writeTasks(tasks);
+    }
+
+    return of(task);
+  }
+
   delete(id: string): Observable<void> {
     const tasks = this.readTasks();
-    const index = tasks.findIndex(task => task.id === id);
+    const index = tasks.findIndex(t => t.id === id);
     if (index !== -1) {
       tasks.splice(index, 1);
       this.writeTasks(tasks);
     }
     return of(null);
   }
 
+  clearCompletedTasks(tasks): void {
+    this.writeTasks(tasks);
```

Comment:
As said would be better to give the tasks parameter a type. 

I would assume some kind of filtering here would be better than doing it in the component and simply save the given tasks here

Reply:
Oh my bad, I missed mentioning the type here. 

Regarding the second point, good input. Thanks. You are right. this can be done at the component level.